### PR TITLE
Update `@metalsmith/permalinks` to 2.5.1

### PR DESCRIPTION
**Pull request by epreston**
Updates the `@metalsmith/permalinks` plugin to 2.5.1

Adjusts metalsmith plugin order and adopts the replacement for the `path` attribute, `permalinks`.

Fixes: #521 

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "metalsmith": "^2.6.1",
     "metalsmith-beautify": "^0.2.0",
     "metalsmith-static": "0.0.5",
-    "node-fetch": "^2.6.7",
+    "node-fetch": "^3.3.2",
```
Comment (willeastcott) [inline package.json:45]: I thought `node-fetch` 3.x wasn't compatible with `require` based JS and only worked in ES Modules:

https://github.com/node-fetch/node-fetch/blob/HEAD/docs/v3-UPGRADE-GUIDE.md#converted-to-es-module

> you are not able to import it with require.

Comment (willeastcott) [reply package.json:45]: Oh, wait, I converted the Metalsmith plugins to modules ages ago....

File: build.mjs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import fs from 'fs';
-import path from 'path';
+import fs from 'node:fs';
+import path from 'node:path';
```
Comment (willeastcott) [inline build.mjs:2]: Just curious - how come these need the `node:` prefix?

File: build.mjs [hunk 3]
```diff
@@ -132,8 +137,10 @@ Metalsmith(__dirname)
         renderer: renderer
     }))
     .use(contents())
+    .use(tutorials('tutorials')())
     .use(permalinks({
-        pattern: ':filename'
+        duplicates: 'error'
+        // pattern: ':filename'
```
Comment (willeastcott) [inline build.mjs:143]: Maybe just delete this line now?

File: package.json (hunk 1 above)
Comment (epreston) [reply package.json:45]: It's used in `index.mjs` like this

```js
import fetch from 'node-fetch';
```

I don't see require based usage in the project.

File: build.mjs (hunk 2 above)
Comment (epreston) [reply build.mjs:2]: Airbags and safety belts. Good habit to avoid name clashes.

File: build.mjs (hunk 3 above)
Comment (epreston) [reply build.mjs:143]: If you are happy with that, sure.  The `duplicates` thing is a handy addition that saved my bacon during testing.

**Review by willeastcott — APPROVED** [review]

File: build.mjs (hunk 3 above)
Comment (epreston) [reply build.mjs]: commented code removed in 8020947

**Comment by willeastcott** [comment]
Wow, it's been a while since this repo had up to date dependencies. You're a life-saver, @epreston! 🙏

**Comment by willeastcott** [comment]
Ooo, I think there might be a problem though. I just did a diff pre and post this PR:

image

This is the tutorial index page.

**Comment by epreston** [comment]
I'll get onto that.

We might need to drop on of the 3rd party plugins for metalsmith that formats the final output.  We could replace it with `prettier` for the same result.  It hasn't gotten updates in 8 years and give security / dependency warnings.

**Comment by epreston** [comment]
I think I know what to do there.

**Comment by epreston** [comment]
Revert ?  This might take an hour or two.

**Comment by willeastcott** [comment]
It's fine. We don't have any deployments planned. 😄

**Comment by epreston** [comment]
I did a winmerge compare after the fix.  There are some differences but they all look like content differences.

**Comment by epreston** [comment]
Looks like the non English languages are getting duplicate navigation menus ?

**Pull request merged**